search-app/src/cmr/search/services/parameters/conversion.clj

```diff
@@ -332,22 +334,40 @@
   (let [values (if (sequential? value) value [value])
         {granule-concept-ids :granule
          collection-concept-ids :collection} (group-by (comp :concept-type cc/parse-concept-id) values)
-        collection-cond (when (seq collection-concept-ids)
-                          (qm/->CollectionQueryCondition
-                           (common-params/parameter->condition
-                            context :collection :concept-id value options)))
-        granule-cond (when (seq granule-concept-ids)
-                       (let [provider-ids (distinct
-                                           (map (comp :provider-id cc/parse-concept-id)
-                                                granule-concept-ids))]
-                         (gc/and-conds [(qm/->CollectionQueryCondition
-                                         (common-params/parameter->condition
-                                          context :collection :provider provider-ids {}))
-                                        (common-params/string-parameter->condition
-                                         concept-type :concept-id granule-concept-ids options)])))]
-    (if (and collection-cond granule-cond)
-      (gc/and-conds [collection-cond granule-cond])
-      (or collection-cond granule-cond))))
+
+        ;; get collection concept ids for granule-concept-ids from metadata-db
+        granules-by-provider (group-by (comp :provider-id cc/parse-concept-id)
```

Comment:
I was thinking we'd only need to do this if collection concept IDs could not be figured out. If a granule concept id is provided for a different collection the set intersection on line 351 is going to remove it.

So I think the logic should be:
```
if-not (seq collection-concept-ids)
...
Figure it out from granule concept-ids
```
That way any time collection concept ids are provided we don't have to query Oracle.